packages/auto-label/src/auto-label.ts

```diff
@@ -211,66 +240,175 @@ handler.autoDetectLabel = (
   )?.github_label;
 };
 
-const BACKFILL_LABEL = 'auto-label:backfill';
+handler.addLabeltoRepoAndIssue = async function addLabeltoRepoAndIssue(
+  owner: string,
+  repo: string,
+  issueNumber: number,
+  context: any,
+  jsonArray: any
+) {
+  const objectInJsonArray = handler.checkIfElementIsInArray(
+    jsonArray,
+    owner,
+    repo
+  );
+
+  let autoDetectedLabelAdded: string | undefined;
+
+  const labelsOnIssue = await handler.checkExistingIssueLabels(
+    context.github,
+    owner,
+    repo,
+    issueNumber
+  );
+
+  if (!objectInJsonArray?.github_label) {
+    console.log(
+      `There was no configured match for the repo ${repo}, trying to auto-detect the right label`
+    );
+    const autoDetectedLabel = handler.autoDetectLabel(
+      jsonArray,
+      context.payload.issue.title
+    );
+    if (autoDetectedLabel) {
+      console.log(
+        `Auto-detected label ${autoDetectedLabel} for ${owner}/${repo}#${issueNumber}`
+      );
+      autoDetectedLabelAdded = await handler.maybeAddAPILabel(
+        context.github,
+        owner,
+        repo,
+        context.payload.issue.number,
+        labelsOnIssue,
+        autoDetectedLabel
+      );
+    } else {
+      console.log(
+        `Auto-detect did not add a label for ${owner}/${repo}#${issueNumber}: '${context.payload.issue.title}'`
+      );
+    }
+    return;
+  }
+
+  const colorNumber = jsonArray.findIndex(
+    (object: JSONData) => objectInJsonArray === object
+  );
+  const githubLabel = objectInJsonArray.github_label;
+
+  const alreadyExists = await handler.checkExistingLabels(
+    context.github,
+    owner,
+    repo,
+    githubLabel
+  );
+
+  if (alreadyExists === null || alreadyExists === undefined) {
+    handler.createLabel(
+      context.github,
+      owner,
+      repo,
+      githubLabel,
+      colorsData[colorNumber].color
+    );
+  } else {
+    console.log(
+      'This label already exists on the repository, will check if it also exists on the issue'
+    );
+  }
+
+  if (labelsOnIssue) {
+    const found = labelsOnIssue.find(
```

Comment:
This is the kind of thing where I'd expect the type of `labelsOnIssue` to have nice static typing, so you don't have to inline the `{name: string}` type you have beow.